container-registry-configuration/README.md

```diff
@@ -0,0 +1,189 @@
+endpoint = ["giantswarm.azurecr.io"]
+
+[plugins."io.containerd.grpc.v1.cri".registry.configs]
+[plugins."io.containerd.grpc.v1.cri".registry.configs."registry-1.docker.io".auth]
+username = "giantswarm-user-account"
+password = "my_token_from_docker_hub"
+```
+
+#### Decision
+
+`1.c` is selected.
+We will use containerd configuration. 
+
+### 2. Authentication
+
+#### 2.a Using unauthenticated accounts
+
+Using only public images without authentication is an option to avoid complexity of securing credentials. However, registry providers have pull/rate limits per account. We will be limited by total of [free account limit](https://docs.docker.com/docker-hub/download-rate-limit/) of Docker Hub (100 pulls per 6 hours per IP address) and [tier limit](https://learn.microsoft.com/en-us/azure/container-registry/container-registry-skus) of Azure Container Registry (3000 ReadOps per minute for standard). Anyone can pull images from our registry and spend our limits, which makes us vulnerable. 
+
+#### 2.b Using authenticated accounts
+
+We can pass credentials to `containerd` configuration to be not limited by registry provider limits.
+
+#### Decision
+
+`2.b` is selected.
+We will use authenticated accounts.
+
+
+### 3. Registries
+
+The infranet page (See `Registry Mirrors`) states this as a desired configuration:
+  - `docker.io` as primary public registry since it has a privilege in docker daemon.
+  - Our own Azure Container Registry as secondary registry because of the security concerns
+
+Since we use `containerd`, docker has not a privelege anymore but there is no need to change this order at the moment. We will follow this configuration for CAPI clusters too.
+
+### 4. Accounts
+
+- 4.a Giantswarm Account + Single account
+- 4.b Giantswarm Account + Per MC including its WCs
+- 4.c Giantswarm Account + Per WC
+- 4.b Customer Accounts
+
+#### Decision
+
+`4.b` is selected.
+
+Since it is a part of the platform itself, we are going to use GiantSwarm accounts in containerd configuration.
+As we do for other operators, we will follow "per MC" approach here to.
+
+### 5. How to provide credentials to WCs
+
+#### 5.a cluster-apps-operator
+
+We can propogate credentials from MC to WC by using `cluster-apps-operator`. It can create a secret per WC with the credentials.
+
+#### 5.b Optional interface + Gitops
+
+We can delegate the responsibility of creating credentials secret to the creator of `cluster-$provider` apps. GitOps can be useful to template the secrets per WC.
+
+#### Decision
+
+`5.b` is selected.
+
+We don't want to add another responsility to `cluster-apps-operator`. 
+
+
+### 6. Default configuration for WCs
+
+- 6.a: Configuring WCs by default
+- 6.b: Not configuring WCs by default
+
+#### Decision
+
+**NOT DECIDED YET**
+
+
+### 7. Where to put the credentials
+
+#### 7.a Customer's Git Repository
+
+We can put the secret into customers' git repositories. Customers will access to our credentials.
```

Comment:
As a note, customers technically have access to these secrets now in vintage

Reply:
As long as we don't disallow the usage of `hostPath` mounts (or `kubectl debug node/<nodeName>`) customers are always able to see all sensitive information on the host filesystem.

Reply:
still jsut for rotation purposes alone, having it outside our jurisdiction feels wrong, why would be put one of our secrets into sth owned by a customer? no matter the security implications

Reply:
It's our registry, which implies that we should be in control of the secret to rotate it.

Reply:
I am all for managing them ourselves.